### ISSUE TITLE
UHD is not in effect because the resolution is considered 1080p

### ISSUE THREAD

**Issue by Liqianyu** [Dis90/plugin.video.discoveryplus#94]
https://github.com/Dis90/plugin.video.discoveryplus/blob/b0ba3642b185489a3db0bba42bc7f898db9fc7bf/resources/lib/dplay.py#L763-L764
I don't know why the above code gets a screen resolution of 1920x1080.

>screenHeight = xbmc.getInfoLabel('System.ScreenHeight')
screenWidth = xbmc.getInfoLabel('System.ScreenWidth')

After modifying to the above code, the resolution is obtained as 3840x2160.

I'm testing on Kodi 21, so I'm not sure if this is a Kodi 21 issue. Maybe an option should be added to force UHD to be enabled by ignoring resolution?

**Comment by Dis90** [comment]
That is not correct fix. We need to find out why Kodi reports wrong resolution. Is your device set to UHD resolution?

**Comment by Dis90** [comment]
What resolutions do you have if you disable UHD from settings? Again, if possible try with another device and also with Kodi 20. Check what resolution Kodi reports in Kodi settings -> System information

With UHD disabled you should have all resolutions up to 1080p.

On InputStream Adaptive add-on settings is option to ignore screen resolution but that is only used for selecting resolution to use when video starts.

**Comment by Dis90** [comment]
I only now noticed that your solution to fix issue was not hardcode resolution.

But both `xbmcgui.getScreenHeight()` and `xbmc.getInfoLabel('System.ScreenHeight')` gives same result in my Mac. Which is Kodi window size and not actual screen resolution.

Can you add following to start of list_menu https://github.com/Dis90/plugin.video.discoveryplus/blob/b0ba3642b185489a3db0bba42bc7f898db9fc7bf/resources/lib/addon.py#L23

```
    import xbmc
    import xbmcgui

    screenHeight = xbmcgui.getScreenHeight()
    screenWidth = xbmcgui.getScreenWidth()
    screenHeight2 = xbmc.getInfoLabel('System.ScreenHeight')
    screenWidth2 = xbmc.getInfoLabel('System.ScreenWidth')

    helper.log('Resolution1')
    helper.log(screenHeight)
    helper.log(screenWidth)

    helper.log('Resolution2')
    helper.log(screenHeight2)
    helper.log(screenWidth2)
```

Then open add-on and find from Kodi log what height and width is reported.

Maybe best option is to hardcode resolutions.

**Comment by Dis90** [comment]
It is strange that those two commands gives different results.

I will hardcode resolutions because commands doesn't output screen resolution when Kodi is in windowed mode although most users uses Kodi in fullscreen.

**Comment by Dis90** [comment]
Yep I was also thinking that xbmcgui method gets GUI resolution and I remember reading somewhere that in some devices it can be different than system resolution. But if getinfolabel outputs correct resolution we can just use it.

Please try if 1440p triggers UHD resolutions if not then we should hardcode resolutions instead of using Kodi/screen resolution.

**Comment by Dis90** [comment]
Yeah I don't know why Kodi doesn't detect screen resolution in window mode. I have to check if there's any alternative way to get screen resolution. If not then maybe best to add override option for resolution to add-on settings or just hardcode 1920x1080 and 3840x2160 for 4k.

**Comment by Dis90** [comment]
Fixed in version 1.7.0
